test/cpp/api/functional.cpp

```diff
@@ -1960,3 +1960,62 @@ TEST_F(FunctionalTest, Dropout3d) {
   ASSERT_TRUE(torch::allclose(input_mean, output.mean(), 0.01, 0.05));
   ASSERT_TRUE((input_std <= output.std()).all().item<bool>());
 }
+
+template<c10::ScalarType S, typename T>
+void test_isfinite(const at::Device& device) {
+  const std::vector<T> values = {
+    std::numeric_limits<T>::lowest(),
+    0, 1, 42,
+    std::numeric_limits<T>::min(),
+    std::numeric_limits<T>::max()
+  };
+  for (const auto value : values) {
+    const auto x = torch::full({3, 3}, value, torch::TensorOptions().dtype(S).device(device));
+    ASSERT_TRUE(torch::isfinite(x).all().template item<bool>());
+  }
+  if (std::numeric_limits<T>::has_infinity) {
+    const auto inf = std::numeric_limits<T>::infinity();
+    const auto x = torch::tensor({
+      -inf,
+      std::numeric_limits<T>::lowest(),
+      static_cast<T>(0),
+      static_cast<T>(1),
+      static_cast<T>(42),
+      std::numeric_limits<T>::min(),
+      std::numeric_limits<T>::max(),
+      inf
+    }, torch::TensorOptions().dtype(S).device(device));
+    ASSERT_TRUE(torch::allclose(
+      torch::isfinite(x).toType(torch::kInt),
```

Comment:
We're taking a BoolTensor, converting to IntTensor, and then comparing it to another IntTensor that was converted from a BoolTensor. Is there a way to just directly compare the BoolTensors ?

Reply:
torch::allclose does not support torch::kBool

Reply:
Could you toss this information into an in-line comment in the testing code?

Reply:
done